src/vcpkg/commands.remove.cpp

```diff
@@ -99,66 +99,49 @@ namespace vcpkg::Remove
             spgh.state = InstallState::NOT_INSTALLED;
             write_update(fs, installed, spgh);
 
-            status_db->insert(std::make_unique<StatusParagraph>(std::move(spgh)));
+            status_db.insert(std::make_unique<StatusParagraph>(std::move(spgh)));
         }
     }
 
-    static void print_plan(const std::map<RemovePlanType, std::vector<const RemovePlanAction*>>& group_by_plan_type)
+    constexpr struct AddressOf
     {
-        static constexpr std::array<RemovePlanType, 2> ORDER = {RemovePlanType::NOT_INSTALLED, RemovePlanType::REMOVE};
-
-        for (const RemovePlanType plan_type : ORDER)
+        template<class T>
+        const T* operator()(const T& t) const
         {
-            const auto it = group_by_plan_type.find(plan_type);
-            if (it == group_by_plan_type.cend())
-            {
-                continue;
-            }
+            return &t;
+        }
+        template<class T>
+        T* operator()(T& t) const
+        {
+            return &t;
+        }
+    } address_of;
 
+    static void print_plan(const RemovePlan& plan)
+    {
+        if (!plan.not_installed.empty())
+        {
+            std::vector<const NotInstalledAction*> not_installed = Util::fmap(plan.not_installed, address_of);
+            Util::sort(not_installed, &BasicAction::compare_by_name);
```

Comment:
As an aside this looks like a place that could use compare_by_name being a functor rather than a function. No change requested.